stringdtype/tests/test_stringdtype.py

```diff
@@ -14,6 +14,11 @@ def string_list():
     return ["abc", "def", "ghi"]
 
 
+@pytest.fixture
+def string_list_similar():
+    return ["left", "right", "leftovers", "righty", "up" "down"]
```

Comment:
Can we get some tests with non-ascii characters?

I also suspect that you'll find when you do this that you could replace your codepoint sorting implementation with just `strcmp` and it will have identical results.

To see why this has to be the case, take a look at [the table](https://en.wikipedia.org/wiki/UTF-8#Encoding) in the wikipedia article on utf-8 encoding. The largest ascii character is `0b01111111`, but the first byte of a multicharacter string will always be at least `0b11000000`. The first bytes are also chosen so that a three-byte character will always sort after a two-byte character, ditto for four-byte strings and three-byte characters. The bits of the codepoint itself are always left in order, although the bits might be split up by continuation bits, but since the continuation bits are always the same, that doesn't affect ordering.

Unfortunately I'm having a heck of a time finding an "official" citation for this fact or even a full explanation for why it works beyond the explanation I just wrote, but I'm pretty sure it's true.

Reply:
You're right - it looks like code points and their corresponding utf8 representations both increase monotonically. I'll swap back to `strcmp` for simplicity, speed, and compactness.